frontend/src/components/accounts/AccountDetails.tsx

```diff
@@ -17,13 +20,50 @@ import WalletLink from "../utils/WalletLink";
 import { Translate } from "react-localize-redux";
 
 export interface Props {
-  account: A.Account;
+  account: Account;
   currentNearNetwork: NearNetwork;
 }
 
+interface State {
+  transactionCount?: AccountStats;
+}
+
 class AccountDetails extends React.Component<Props> {
+  state: State = {
+    transactionCount: {},
+  };
+
+  collectTransactionCount = async () => {
+    new AccountsApi()
+      .queryTransactionCount(this.props.account.accountId)
+      .then((transactionCount) => {
+        if (transactionCount) {
+          this.setState({ transactionCount });
+        }
+        return;
+      })
+      .catch((error) => console.error(error));
+  };
+
+  componentDidMount() {
+    this.collectTransactionCount();
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (prevProps.account.accountId !== this.props.account.accountId) {
+      this.setState({ transactionCount: {} });
+      this.collectTransactionCount();
```

Comment:
1. set the `transactionCount` to undefined
2. use the `setState` callback to avoid a situation when the `transactionCount` get reset after `collectTransactionCount` set state to the fresh values. It should not happen in this scenario since setting a state is faster than a query to the network, but it is better to use the proper setup always to avoid unexpected bugs due to ordering of certain events.

```suggestion
      this.setState({ transactionCount: undefined }, this.collectTransactionCount);
```